### ISSUE TITLE
linter: no-redeclare false positives when variable and type have same name

### ISSUE THREAD

**Issue by EYHN** [oxc-project/oxc#8237]
### What version of Oxlint are you using?

0.15.5

### What command did you run?

oxlint -c oxlint.json

### What does your `.oxlint.json` config file look like?

```
{
  "rules": {
    "no-redeclare": "error"
  }
}
```

### What happened?

with code

```ts
export const a = {
  hello: "world",
};

export type a = typeof a;
```

oxlint shows error

```
  × eslint(no-redeclare): 'a' is already defined.
   ╭─[src\bad.ts:1:14]
 1 │ export const a = {
   ·              ┬
   ·              ╰── 'a' is already defined.
 2 │   hello: "world",
   ╰────
   ╭─[src\bad.ts:5:13]
 4 │
 5 │ export type a = typeof a;
   ·             ┬
   ·             ╰── It can not be redeclare here.
   ╰────
```

Having both type and variable with same name is allowed in typescript and eslint not show this error, so I think it is a bug

**Comment by Sysix** [comment]
**info in general:**
we are (currently) supporting only the eslint version:

https://github.com/oxc-project/oxc/blob/c90fc16bba829862ffa3f6e79368dc1e5ed63898/crates/oxc_linter/src/rules/eslint/no_redeclare.rs#L29-L31

The ts rules has a extra option which defaults to true:
https://typescript-eslint.io/rules/no-redeclare/#options

**Comment by shulaoda** [comment]
Actually, `eslint` only supports diagnostics for JS syntax, while `typescript-eslint` will default to reporting an error for `type` redeclaration.
